### PR TITLE
More warning fixes

### PR DESCRIPTION
**Pull request by bje-**
Nothing controversial here, I suspect!

### CODE REVIEW

File: src/easycurl.cpp [hunk 1]
```diff
@@ -467,24 +467,6 @@ void wxEasyCurl::SetProxyAddress(const wxString &proxy)
 	gs_curlProxyAddress = proxy;
 }
 
-static std::string wstr2str(const wchar_t *s)
```
Comment (nickdiorio) [inline src/easycurl.cpp]: It seems as if this function is used on line 533 of the same file, and that removing this function causes an unsolved reference.  Is this function defined elsewhere?

File: src/jsonval.cpp [hunk 2]
```diff
@@ -3008,8 +3008,7 @@ wxJSONValue::DeepCopy(const wxJSONValue& other)
 wxJSONRefData*
 wxJSONValue::GetRefData() const
 {
-	wxJSONRefData* data = m_refData;
-	return data;
+	return m_refData;
```
Comment (nickdiorio) [inline src/jsonval.cpp]: Agreed, this is a fairly pointless intermediate variable

**Comment by nickdiorio** [comment]
I merged in three of the commits, but commit c26e5c2 still needs resolved before it can be merged.

File: src/easycurl.cpp (hunk 1 above)
Comment (bje-) [reply src/easycurl.cpp]: Sorry, you're right. It's only used on Windows. I've just pushed a new change that (as minimally as possible) eliminates wstr2str() by initialising a std::string with iterators. If you agree, the rest of the changes on this branch should now be able to be merged.

**Pull request closed without merging**